update: changed Sign to use OCI image manifest as default (#573)
notaryproject/notation: specs/commandline/sign.md

       --plain-http                 registry access via plain HTTP
       --plugin-config stringArray  {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
       --signature-format string    signature envelope format, options: "jws", "cose" (default "jws")
       --signature-manifest string  manifest type for signature, options: "image", "artifact" (default "image")
  -u,  --username string            username for registry operations (default to $NOTATION_USERNAME if not specified)
  -m,  --user-metadata stringArray  {key}={value} pairs that are added to the signature payload
  -v,  --verbose                    verbose mode
```

## Use OCI image manifest to store signatures

> By default, Notation uses [OCI image manifest][oci-image-spec] to store signatures in registries. Users could instead choose [OCI artifact manifest][oci-artifact-manifest] by enabling the `--signature-manifest artifact` flag. In this case, the registry is REQUIRED to support `OCI artifact` and with [Referrers API][oci-referers-api] enabled.

Reviewer: ```suggestion
By default, Notation uses [OCI image manifest][oci-image-spec] to store signatures in registries. Users can use [OCI artifact manifest][oci-artifact-manifest] by enabling the `--signature-manifest artifact` flag. When using OCI artifact manifest to store the signature, the registry is REQUIRED to support both `OCI artifact` and [Referrers API][oci-referers-api].
```

Note that there is no deterministic way to determine whether a registry supports `OCI artifact` or not. The following response status contained in error messages MAY indicate that the registry doesn't support `OCI artifact`.

- Response status `400 BAD Request` with error code `MANIFEST_INVALID` or `UNSUPPORTED`

### Set config property for OCI image manifest

> OCI image manifest requires additional property `config` of type `descriptor`, which is not required by OCI artifact manifest. On default signing with OCI image manifest, Notation uses empty JSON object `{}` as the default configuration content, and thus the `config` property is fixed, as following:

Reviewer: ```suggestion
OCI image manifest requires additional property `config` of type `descriptor`, which is not required by OCI artifact manifest. When signing with OCI image manifest, Notation uses empty JSON object `{}` as the default configuration content, and thus the `config` property is fixed, as following:
```

Successfully signed localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
```

> ### Sign an artifact and store the signature using OCI artifact manifest

Reviewer 2: I think we can remove this usage.

Reviewer 3: I think we can keep it for now and evaluate later since we still support this case.
